source/MRMesh/MRVoxelsConversions.cpp

```diff
@@ -474,74 +489,185 @@ TriangulationPlan{0, 3, 8},
+    }
+    else if ( iterIndex == 6 )
+    {
+        if ( ( config & ( 1 << cMapNeighborsShift[6] ) ) != ( config & ( 1 << cMapNeighborsShift[7] ) ) )
+            return true;
+        return false;
+    }
+    return false;
+}
+
 }
 
 using namespace MarchingCubesHelper;
 using ConstAccessor = openvdb::FloatGrid::ConstAccessor;
 using VdbCoord = openvdb::Coord;
 
-Vector3f voxelPositionerLinear( const Vector3f& pos0, const Vector3f& pos1, float v0, float v1, float iso )
+inline Vector3f voxelPositionerLinearInline( const Vector3f& pos0, const Vector3f& pos1, float v0, float v1, float iso )
 {
-    auto ratio = std::clamp( std::abs( iso - v0 ) / std::abs( v1 - v0 ), 0.0f, 1.0f );
+    const auto ratio = std::clamp( std::abs( iso - v0 ) / std::abs( v1 - v0 ), 0.0f, 1.0f );
     return ( 1.0f - ratio ) * pos0 + ratio * pos1;
 }
 
-SeparationPoint findSeparationPoint( const VdbVolume& volume, const ConstAccessor& acc, const VdbCoord& minCoord,
-    const VolumeIndexer& indexer, VoxelId base, NeighborDir dir, const VolumeToMeshParams& params )
+Vector3f voxelPositionerLinear( const Vector3f& pos0, const Vector3f& pos1, float v0, float v1, float iso )
+{
+    return voxelPositionerLinearInline( pos0, pos1, v0, v1, iso );
+}
+
+class UseDefaultVoxelPointPositioner {};
+class OmitNaNCheck {};
+
+template <typename Args>
+bool findSeparationPoint( SeparationPoint& sp, const VdbVolume& volume, const ConstAccessor& acc,
+                          const openvdb::Coord& coord, const Vector3i& basePos, float valueB, NeighborDir dir,
+                          const VolumeToMeshParams& params )
 {
-    auto basePos = indexer.toPos( base );
     if ( basePos[int( dir )] + 1 >= volume.dims[int( dir )] )
-        return {};
-    auto coord = openvdb::Coord{ basePos.x + minCoord.x(),basePos.y + minCoord.y(),basePos.z + minCoord.z() };
-    const auto& valueB = acc.getValue( coord );// volume.data[base];
+        return false;
     auto nextCoord = coord;
     nextCoord[int( dir )] += 1;
-    const auto& valueD = acc.getValue( nextCoord );// volume.data[nextId];
+    float valueD = acc.getValue( nextCoord );// volume.data[nextId];
+
     bool bLower = valueB < params.iso;
     bool dLower = valueD < params.iso;
-
     if ( bLower == dLower )
-        return {};
+        return false;
 
-    SeparationPoint res;
     Vector3f coordF = Vector3f( float( coord.x() ), float( coord.y() ), float( coord.z() ) );
     Vector3f nextCoordF = Vector3f( float( nextCoord.x() ), float( nextCoord.y() ), float( nextCoord.z() ) );
     auto bPos = params.origin + mult( volume.voxelSize, coordF );
     auto dPos = params.origin + mult( volume.voxelSize, nextCoordF );
-    res.position = params.positioner( bPos, dPos, valueB, valueD, params.iso );
-    res.vid = VertId{ 0 };
-    return res;
+    if constexpr ( boost::mp11::mp_contains<Args, UseDefaultVoxelPointPositioner>::value )
+        sp.position = voxelPositionerLinearInline( bPos, dPos, valueB, valueD, params.iso );
```

Comment:
Is it much slower to call this function via `params.positioner`?

Reply:
The overall impact is up to 10%.